imagedephi/gui/utils/image.py

```diff
@@ -0,0 +1,91 @@
+from io import BytesIO
+from typing import TYPE_CHECKING
+
+from PIL import Image, UnidentifiedImageError
+from fastapi.responses import StreamingResponse
+import tifftools
+
+from imagedephi.gui.utils.constants import MAX_ASSOCIATED_IMAGE_HEIGHT
+
+if TYPE_CHECKING:
+    from tifftools.tifftools import IFD
+
+
+def extract_thumbnail_from_image_bytes(ifd, file_name: str) -> Image.Image | None:
```

Comment:
this should be:
`def extract_thumbnail_from_image_bytes(ifd: IFD, file_name: str) -> Image.Image | None:`

but having the `IFD` type hint throws a NameError(?!)

Any insight into how to resolve this would be really appreciated. I can't figure out why python cares about the typehint only here.

@manthey @naglepuff